micro-rdk/src/common/conn/viam.rs

```diff
@@ -573,48 +575,20 @@ where
             log::error!("couldn't store the robot configuration reason {:?}", err);
         }
 
+        #[cfg(feature = "esp32")]
+        let hook = || crate::esp32::esp_idf_svc::hal::reset::restart();
+        #[cfg(not(feature = "esp32"))]
+        let hook = || std::process::exit(0);
```

Comment:
- Before we just called `std::process::exit`, but this goes back to an older style where on esp32 we do `hal::reset::restart`. What was the motivation for that change. I think I know, but I'd like to be sure.
- If this is to effect the change I suspect, I think I'd want to see the same change in the `RestartMonitor`.
- A block nearly identical to this exists in `grpc::robot_shutdown`. Can we DRY this?

Reply:
the motivation was to get rid of the coredump we see everytime on config change restarts since I'm already modifying the config_monitor's behavior. 

I mention it in the description about partially addressing [RSDK-9594](https://viam.atlassian.net/browse/RSDK-9594), but just in config monitor.

I can look into `grpc::robot_shutdown`, but that depends if you want that done across the repo in RSDK-9594 or just here. Which would you prefer?

Reply:
per offline discussion, changing this back to just use `std::process::exit`. the right changes will be done in RSDK-9594